### PR TITLE
Feature/top up

### PR DESCRIPTION
**Pull request by Alphonce-Mutebi**
+ Integrated agentAcceptDepositTransaction contract calls
+ Integrated agentAcceptWithdrawalTransaction contract calls

### CODE REVIEW

**Review by steffenkrogmann — APPROVED** [review]

File: src/screens/TransactionRequests/TransactionConfirmationScreen.tsx [hunk 1]
```diff
@@ -1,25 +1,246 @@
...
+  const operation = "TopUp";
+  const transaction = route.params?.transaction;
+  const dispatch = useDispatch();
+
+  const [isActionSuccess, setIsActionSuccess] = useState(true);
+  const [isLoading, setIsLoading] = useState(true);
+  const [loadingMessage, setLoadingMessage] = useState("");
+  // const publicAddress =
+  //   WakalaContractKit.getInstance().userMetadata.publicAddress;
+  const publicAddress = "0x9FDf3F87CbEE162DC4a9BC9673E5Bb6716186757";
```
Comment (Allen-Muhani) [inline src/screens/TransactionRequests/TransactionConfirmationScreen.tsx]: Why is this hardocded?

File: src/screens/TransactionRequests/TransactionConfirmationScreen.tsx [hunk 2]
```diff
@@ -1,25 +1,246 @@
...
+  errorImage: {
+    height: 180,
+    maxWidth: SIZES.width * 0.8,
+    resizeMode: "contain",
+    marginBottom: 20,
+  },
+
+  title: {
+    fontSize: 16,
+    lineHeight: 24,
```
Comment (Allen-Muhani) [inline src/screens/TransactionRequests/TransactionConfirmationScreen.tsx]: use centralized text styling.

File: src/screens/TransactionRequests/TransactionSuccess.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    color: COLORS.textPrimary,
+    fontFamily: FONTS.headline.fontFamily,
+    fontSize: FONTS.headline.fontSize,
+    lineHeight: FONTS.headline.lineHeight,
+    marginTop: 15,
+  },
+  cardHeader: {
+    fontFamily: FONTS.headline.fontFamily,
+    fontSize: FONTS.headline.fontSize,
+    lineHeight: FONTS.headline.lineHeight,
```
Comment (Allen-Muhani) [inline src/screens/TransactionRequests/TransactionSuccess.tsx]: All the first three lines of code should be one line of code.

File: src/screens/TransactionRequests/TopUpViewRequestScreen.tsx [hunk 4]
```diff
@@ -1,104 +1,311 @@
...
+  errorImage: {
+    height: 180,
+    maxWidth: SIZES.width * 0.8,
+    resizeMode: "contain",
+    marginBottom: 20,
+  },
+
+  title: {
+    fontSize: 16,
+    lineHeight: 24,
```
Comment (Allen-Muhani) [inline src/screens/TransactionRequests/TopUpViewRequestScreen.tsx]: use entralized styling text.

File: src/screens/TransactionRequests/TopUpViewRequestScreen.tsx [hunk 5]
```diff
@@ -1,104 +1,311 @@
...
+    color: "#333333",
+    textAlign: "center",
+    fontFamily: "Rubik_500Medium",
+  },
+
+  text: {
+    fontSize: 14,
+    lineHeight: 21,
+    color: "#333333",
```
Comment (Allen-Muhani) [inline src/screens/TransactionRequests/TopUpViewRequestScreen.tsx]: use entralized styling text.

File: src/screens/TransactionRequests/TopUpViewRequestScreen.tsx [hunk 6]
```diff
@@ -1,104 +1,311 @@
...
+    textAlign: "center",
+    fontFamily: "Rubik_400Regular",
+    marginTop: 20,
+  },
+
+  button: {
+    fontSize: 20,
+    lineHeight: 24,
+    color: "#133FDB",
```
Comment (Allen-Muhani) [inline src/screens/TransactionRequests/TopUpViewRequestScreen.tsx]: use centralized styling text and color.

File: src/screens/TransactionRequests/TransactionConfirmationScreen.tsx [hunk 7]
```diff
@@ -1,25 +1,246 @@
...
+    color: "#333333",
+    textAlign: "center",
+    fontFamily: "Rubik_500Medium",
+  },
+
+  text: {
+    fontSize: 14,
+    lineHeight: 21,
+    color: "#333333",
```
Comment (Allen-Muhani) [inline src/screens/TransactionRequests/TransactionConfirmationScreen.tsx]: use centralized text styling.

File: src/screens/TransactionRequests/TransactionConfirmationScreen.tsx [hunk 8]
```diff
@@ -1,25 +1,246 @@
...
+    textAlign: "center",
+    fontFamily: "Rubik_400Regular",
+    marginTop: 25,
+  },
+
+  button: {
+    fontSize: 20,
+    lineHeight: 24,
+    color: "#133FDB",
```
Comment (Allen-Muhani) [inline src/screens/TransactionRequests/TransactionConfirmationScreen.tsx]: use centralized text styling.

File: src/screens/TransactionRequests/TransactionSuccess.tsx [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+  },
+  imageContainer: {
+    flexDirection: "row",
+    marginTop: 15,
+    justifyContent: "center",
+  },
+  textContainer: {
+    textAlign: "center",
+    justifyContent: "center",
+    fontFamily: FONTS.displayBold.fontFamily,
```
Comment (Allen-Muhani) [inline src/screens/TransactionRequests/TransactionSuccess.tsx:79]: use centralized text styling.

File: src/screens/TransactionRequests/TransactionSuccess.tsx [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+    color: COLORS.primary,
+    fontFamily: FONTS.body3.fontFamily,
+    fontSize: FONTS.body3.fontSize,
+    lineHeight: FONTS.body3.lineHeight,
+    marginTop: 10,
+  },
+  subHeadingText: {
+    color: COLORS.textPrimary,
+    fontFamily: FONTS.headline.fontFamily,
+    fontSize: FONTS.headline.fontSize,
```
Comment (Allen-Muhani) [inline src/screens/TransactionRequests/TransactionSuccess.tsx:108]: use centralized text styling.

File: src/screens/TransactionRequests/TransactionSuccess.tsx [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+    borderBottomEndRadius: 10,
+    borderBottomLeftRadius: 10,
+    borderBottomStartRadius: 10,
+    borderTopEndRadius: 10,
+    borderTopStartRadius: 10,
+    marginTop: 100,
+  },
+  headingText: {
+    color: COLORS.primary,
+    fontFamily: FONTS.body3.fontFamily,
```
Comment (Allen-Muhani) [inline src/screens/TransactionRequests/TransactionSuccess.tsx:100]: use centralized text styling.

File: src/screens/TransactionRequests/TransactionSuccess.tsx [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+  },
+  cardHeader: {
+    fontFamily: FONTS.headline.fontFamily,
+    fontSize: FONTS.headline.fontSize,
+    lineHeight: FONTS.headline.lineHeight,
+    textAlign: "center",
+    color: COLORS.textPrimary,
+  },
+  textButton: {
+    fontFamily: FONTS.sh2.fontFamily,
```
Comment (Allen-Muhani) [inline src/screens/TransactionRequests/TransactionSuccess.tsx:114]: use centralized text styling.

File: src/screens/TransactionRequests/TransactionConfirmationScreen.tsx [hunk 13]
```diff
@@ -1,25 +1,246 @@
...
+            </Text>
+
+            <TouchableOpacity onPress={() => props.handleAction()}>
+              <Text style={modalStyles.button}>Got it!</Text>
+            </TouchableOpacity>
+          </View>
+        ) : (
+          <View>
+            <Ionicons
+              name="checkmark-circle"
```
Comment (Allen-Muhani) [inline src/screens/TransactionRequests/TransactionConfirmationScreen.tsx:44]: use centralized color styling.

File: src/screens/TransactionRequests/TopUpViewRequestScreen.tsx [hunk 14]
```diff
@@ -1,104 +1,311 @@
...
+      >
         {!processIsError ? (
-            <View style={styles.modalOverlay}>
-                <LargeModal 
-                    onPressHanlder={() => setModalVisible(false)}
-                    title={"Oh Snap!"}
-                    message={"Something just happened. Please try  again."}
-                    btnText={"Try again"}
-                ></LargeModal>
-            </View>
```
Comment (Allen-Muhani) [inline src/screens/TransactionRequests/TopUpViewRequestScreen.tsx:72]: Why not use the large modal file as utils?

**Review by Allen-Muhani — CHANGES_REQUESTED** [review]

**Review by Allen-Muhani — APPROVED** [review]

**Pull request merged**